product_gtin/models/product_product.py

```diff
@@ -0,0 +1,17 @@
+# Copyright 2004-2011 Tiny SPRL (<http://tiny.be>)
+# Copyright 2010-2011 Camptocamp Austria (<http://www.camptocamp.at>)
+# Copyright 2016 ACSONE SA/NV (<http://acsone.eu>)
+# Copyright 2021 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openerp import api, models
+
+
+class ProductProduct(models.Model):
+    _name = "product.product"
+    _inherit = ["abstract.barcodenumber", "product.product"]
+    _barcodes = ["barcode"]
```

Comment:
This is not serving for anything, but we can give it an utility:

- Move it to the mixin
- Put it a proper variable name:  `_barcode_fields = ["barcode"]` (or if it gets more complicated, do it a single field)
- Move the check to the mixin, and see if you can put on the `@api.constrains` the variable as arguments.
- On the inherited model, you only need to change the variable if not matching the field name in the model.